entities/shop/BillingCounter.java

```diff
@@ -89,4 +91,24 @@ public static float getWeightOfItemInCart(Item item) {
         }
         return 1;
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        BillingCounter that = (BillingCounter) o;
+        return counterNum == that.counterNum && Objects.equals(employee, that.employee);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(counterNum, employee);
+    }
+
+    @Override
+    public int compareTo(Object o) {
+        BillingCounter counter = (BillingCounter) o;
+        if (this == counter) return 0;
+        return (counterNum == counter.counterNum && Objects.equals(employee, counter.employee)) ? 0 : this.hashCode() - counter.hashCode();
```

Comment:
Please let me know if this is the right implementation for compareTo ? As per my understanding this method is called when a new object is added to the set to avoid duplicates. So if they are equal then its zero but, how to know what values to return in other cases.

Reply:
CompareTo is implemented for comparimg the counter number.